src/Framework/PaymentGateways/Types/OffSitePaymentGateway.php

```diff
@@ -2,14 +2,53 @@
 
 namespace Give\Framework\PaymentGateways\Types;
 
+use Give\Framework\PaymentGateways\Commands\PaymentCancelled;
+use Give\Framework\PaymentGateways\Commands\PaymentCommand;
+use Give\Framework\PaymentGateways\Commands\PaymentFailed;
+use Give\Framework\PaymentGateways\Commands\PaymentProcessing;
 use Give\Framework\PaymentGateways\Contracts\OffsiteGatewayInterface;
 use Give\Framework\PaymentGateways\PaymentGateway;
-use Give\Framework\PaymentGateways\Traits\OffsiteGateway;
 
 /**
  * @since 2.18.0
  */
 abstract class OffSitePaymentGateway extends PaymentGateway implements OffsiteGatewayInterface
 {
-    use OffsiteGateway;
+    /**
+     * @unreleased
+     *
+     * @param int $donationId
+     *
+     * @return PaymentCommand
+     */
+    public function returnSuccessFromOffsiteRedirect($donationId)
+    {
+        return new PaymentProcessing();
+    }
+
+    /**
+     * Handle failed donation redirect.
+     *
+     * @unreleased
+     *
+     * @param int $donationId
+     *
+     * @return PaymentCommand
+     */
+    public function returnFailureFromOffsiteRedirect($donationId)
+    {
+        return new PaymentFailed();
+    }
+
+    /**
+     * Handle cancelled donation redirect.
+     *
+     * @param int $donationId
+     *
+     * @return PaymentCancelled
+     */
+    public function returnCancelFromOffsiteRedirect($donationId)
+    {
+        return new PaymentCancelled();
+    }
```

Comment:
@ravinderk there was a reason we had an `OffsiteGateway` trait. It's because a gateway should have the flexibility to extend OffSitePaymentGateway, or implement OffsiteGatewayInterface themselves and pull in the trait for the helpers.

Also same comment about moving away from all payment status commands.  We need to talk through this.